api-gateway/src/main/resources/static/scripts/auth/reset-pwd-form/reset-pwd-form.template.html

```diff
@@ -24,12 +28,78 @@ <h3 class="text-danger text-center" ng-repeat="error in errorMessages">{{ error
           }
         }
       </script>
-
-      <span  class="help-block">Password is required.</span>
+      <span class="help-block">Password is required.</span>
     </div>
 
     <div class="group-form d-flex justify-content-center">
       <button id="button" class="btn btn-default" type="submit">Reset Password</button>
     </div>
   </form>
-</div>
\ No newline at end of file
+</div>
+
+
+<script>
+  const passwordInput = document.getElementById("pwd");
+  const passwordStrength = document.getElementById("password-strength");
+
+  passwordInput.addEventListener("input", updatePasswordStrength);
+
+  function updatePasswordStrength() {
+    const password = passwordInput.value;
+    const strength = calculatePasswordStrength(password);
+    const strengthText = getStrengthText(strength);
+
+    passwordStrength.textContent = strengthText;
+    passwordStrength.className = `password-strength strength-${strength}`;
+  }
+
+  function calculatePasswordStrength(password) {
+
+    const pattern = /^(?=.*[0-9])(?=.*[a-z])(?=.*[A-Z])(?=.*[@#$%^&+=*!()_.<>,{}])(?=\S+$).{8,}$/;
+
+    if (pattern.test(password)) {
```

Comment:
Can you explain what .test does ?

Reply:
It's a method used on a regular expression object to check if the string I'm giving it matches the regular expression pattern I am specifying. In this case, it is checking to see if the password matches the regex.